pkg/internal/targz.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// TarGZ makes archives the source as the dstination filename. This avoids a large (~3MB) dependency on archiver/v3.
+func TarGZ(dst, src string) error { // dst, src order like io.Copy
```

Comment:
note I'm using implicit test coverage for now, and when I looked at it, the places not touched will be hard, if possible to reproduce (ex tar err when the input is a unix pipe). However, I'm fine making unit tests for this anyway if we want to even if it doesn't improve statement coverage.